dhall-docs/src/Dhall/Docs.hs

```diff
@@ -105,16 +107,25 @@ defaultMain = \case
         GHC.IO.Encoding.setLocaleEncoding System.IO.utf8
         resolvedPackageDir <- Path.IO.resolveDir' packageDir
 
-        outLinkExists <- System.Directory.doesDirectoryExist docLink
-        Control.Monad.when outLinkExists $ System.Directory.removeFile docLink
+        outDirExists <- System.Directory.doesPathExist docLink
+        Control.Monad.when outDirExists $ do
+            isLink <- System.Directory.pathIsSymbolicLink docLink
+            if isLink then System.Directory.removeFile docLink
+            else die $ "The specified --output-link (" <> docLink <> ") already "
+                    <> "exists and its not a symlink. The --output-link flag "
+                    <> "needs to either not exist or to be a symlink"
 
         resolvedDocLink <- Path.IO.resolveDir' docLink
-
         let packageName = resolvePackageName resolvedPackageDir
         generateDocs resolvedPackageDir resolvedDocLink packageName
     Version ->
         putStrLn (showVersion Meta.version)
 
+die :: String -> IO a
```

Comment:
Nit: I think there are benefits to having a single dominant string type in this code base, and it should probably be `Text .